src/OpenLoco/Paint/Paint.cpp

```diff
@@ -63,20 +65,85 @@ namespace OpenLoco::Paint
     // 0x0048DDE4
     [[nodiscard]] InteractionArg PaintSession::getStationNameInteractionInfo(const uint32_t flags)
     {
-        _getMapCoordinatesFromPosFlags = flags;
-        registers regs;
-        regs.edi = reinterpret_cast<uint32_t>(*_dpi);
-        call(0x0048DDE4, regs);
-        return InteractionArg{ _sessionInteractionInfoX, _sessionInteractionInfoY, { _sessionInteractionInfoValue }, _sessionInteractionInfoType, _sessionInteractionInfoBh };
+        InteractionArg interaction{};
+
+        // -2 as there are two interaction items that you can't filter out adjust in future
+        if (flags & (1 << (static_cast<uint32_t>(InteractionItem::station) - 2)))
+        {
+            return interaction;
+        }
+
+        auto zoom = (*_dpi)->zoom_level;
+        auto left = (*_dpi)->x >> zoom;
+        auto top = (*_dpi)->y >> zoom;
+        auto right = ((*_dpi)->width >> zoom) + left;
+        auto bottom = ((*_dpi)->height >> zoom) + top;
+        auto rect = Ui::Rect::fromLTRB(left, top, right, bottom);
```

Comment:
This block of code looks like it could be refactored into a re-useable member of `dpi`.

Reply:
I'm not sure its really used much. But if you think it would be useful can add something like that.